src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -90,7 +90,32 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		@IWorkspaceConfigurationService private _workspaceConfigurationService: IWorkspaceConfigurationService,
 		@IClipboardService private _clipboardService: IClipboardService,
 		@ICommandService private _commandService: ICommandService
-	) { }
+	) {
+		this.initializeConnectionProviders();
+	}
+
+	private initializeConnectionProviders() {
+		this.setConnectionProviders();
+		if (this._capabilitiesService) {
+			this._capabilitiesService.onCapabilitiesRegistered(() => {
+				this.setConnectionProviders();
+				if (this._connectionDialog) {
+					this._connectionDialog.updateConnectionProviders(this._providerTypes, this._providerNameToDisplayNameMap);
+				}
+			});
+		}
+	}
+
+	private setConnectionProviders() {
```

Comment:
would be nice to have comments for these methods

Reply:
sure